pangeo/values.yaml

```diff
@@ -56,25 +56,52 @@ jupyterhub:
 dask-gateway:
   # dask-gateway configuration goes here
   # See https://github.com/dask/dask-gateway/blob/master/resources/helm/dask-gateway/values.yaml
+  # The values here are optimized for deployment within a jupyterhub.
   gateway:
-    clusterManager:
-      image:
-        name: pangeo/base-notebook
-        tag: 2019.09.23
+    prefix: "/services/dask-gateway"
+    auth:
+      type: jupyterhub
 
-      clusterStartTimeout: 600
-      workerStartTimeout: 600
-      worker:
+    backend:
+      scheduler:
         extraPodConfig:
           tolerations:
+            - key: "k8s.dask.org/dedicated"
+              operator: "Equal"
+              value: "scheduler"
+              effect: "NoSchedule"
             - key: "k8s.dask.org_dedicated"
               operator: "Equal"
-              value: "worker"
-      scheduler:
+              value: "scheduler"
+              effect: "NoSchedule"
+      worker:
         extraPodConfig:
           tolerations:
+            - key: "k8s.dask.org/dedicated"
+              operator: "Equal"
+              value: "worker"
+              effect: "NoSchedule"
             - key: "k8s.dask.org_dedicated"
               operator: "Equal"
               value: "worker"
-    auth:
-      type: jupyterhub
+              effect: "NoSchedule"
+
+    extraConfig:
+      # Use the mapping form, to support merging multiple values.yaml
+      optionHandler: |
+        from dask_gateway_server.options import Options, Integer, Float, String
+        def option_handler(options):
+            if ":" not in options.image:
+                raise ValueError("When specifying an image you must also provide a tag")
+            return {
+                "worker_cores_limit": options.worker_cores,
+                "worker_cores": min(options.worker_cores / 2, 1),
+                "worker_memory": "%fG" % options.worker_memory,
+                "image": options.image,
+            }
+        c.Backend.cluster_options = Options(
+            Integer("worker_cores", 2, min=1, max=4, label="Worker Cores"),
+            Float("worker_memory", 4, min=1, max=8, label="Worker Memory (GiB)"),
+            String("image", default="{JUPYTER_IMAGE_SPEC}", label="Image"),
```

Comment:
@jcrist are these config options ever executed on the `dask-gateway` server, or is it just client side? The environment variable `{JUPYTER_IMAGE_SPEC}` is set by jupyterhub and will always be present on our clients, but not on the server.

Reply:
The options configured here are always executed on the server. The client can set default options in their dask yaml config files, which *can* be templated by environment variables. So having a default of `{JUPYTER_IMAGE_SPEC}` doesn't make sense on the server side, but having the following config on the user's side:

```yaml
gateway:
  cluster:
    options:
      image: "{JUPYTER_IMAGE_SPEC}"
```

would make sense.

See https://gateway.dask.org/configuration-user.html#default-configuration for full user-side config reference.

Reply:
Thanks for the clarification!